.github/workflows/tests.yaml

```diff
@@ -139,20 +143,26 @@ jobs:
           CI: true
         run: bin/run-browser-tests-ci --shard=${{ matrix.shard_number }}/${{strategy.job-total}}
 
+      # We'll treat aws as the source of truth for images snapshots. Attempts to run both will
+      # end up with two places trying to update the same images.
       - name: Verify no new image snapshots added
         if: ${{ matrix.cloud == 'aws' }}
         # jest-image-snapshots automatically adds snapshot if it's missing.
         # Use git diff to fail if we detect new images.
         run: git add browser-test/image_snapshots ; git diff --compact-summary --exit-code HEAD
 
+      # Upload image diff files, these are the single image containing the before, after, and diff parts.
+      # We want this for aws and azure so we have access to what happened in failed tests.
       - name: Upload image diff outputs
-        if: ${{ matrix.cloud == 'aws' && failure() }}
+        if: failure()
```

Comment:
@rockycodes took a second pass to make sure I'm upload the things that make sense. Decided to include the image diffs as we'll want them for quick review and they aren't stored back into the repo like the image snapshots are.